src/parser/WebVTT.cpp

```diff
@@ -18,8 +18,27 @@
 
 #include "WebVTT.h"
 #include "../helpers.h"
+#include "../log.h"
 #include <cstring>
 
+constexpr const char* signatureCharsBOM = "\xEF\xBB\xBF\x57\x45\x42\x56\x54\x54";
+constexpr const char* signatureChars = "\x57\x45\x42\x56\x54\x54";
+constexpr char signatureLastChars[] = {'\x0A', '\x0D', '\x20', '\x09'};
+
+static bool ValidateSignature(const char* data, const char* signature)
+{
+  if (strlen(data) > strlen(signature))
+  {
+    if (strncmp(data, signature, strlen(signature)) == 0)
+    {
+      // Check if last char is valid
+      if (strchr(signatureLastChars, data[strlen(signature)]) != NULL)
```

Comment:
All of the functions should be called C++ way, so fully qualified `std::strlen` etc.